packages/calendar-utils/src/calendar-body/calendar-body.styles.ts

```diff
@@ -124,42 +133,73 @@ const getInputFontColor = (props: TCalendarBody) => {
   }
   return designTokens.fontColorForInput;
 };
-const getInputContainerStyles = (props: TCalendarBody, state: TState) => {
-  return css`
-    appearance: none;
-    background-color: ${props.isDisabled
-      ? designTokens.backgroundColorForInputWhenDisabled
-      : designTokens.backgroundColorForInput};
-    border: 1px solid ${getInputBorderColor(props, state)};
-    border-radius: ${designTokens.borderRadiusForInput};
-    box-sizing: border-box;
-    color: ${getInputFontColor(props)};
-    cursor: ${props.isDisabled ? 'not-allowed' : 'default'};
-    width: 100%;
-    height: ${designTokens.sizeHeightInput};
-    align-items: center;
-    display: flex;
-    font-size: ${designTokens.fontSizeDefault};
-    font-family: inherit;
-    min-width: ${designTokens.constraint5};
-    transition: border-color ${designTokens.transitionStandard},
-      box-shadow ${designTokens.transitionStandard};
+const getInputContainerStyles = (
+  props: TCalendarBody,
+  state: TState,
+  isNewTheme: boolean
+) => {
+  return [
+    css`
+      appearance: none;
+      background-color: ${props.isDisabled
+        ? designTokens.backgroundColorForInputWhenDisabled
+        : designTokens.backgroundColorForInput};
+      border: 1px solid ${getInputBorderColor(props, state)};
+      border-radius: ${designTokens.borderRadiusForInput};
+      box-sizing: border-box;
+      color: ${getInputFontColor(props)};
+      cursor: ${props.isDisabled ? 'not-allowed' : 'default'};
+      width: 100%;
+      height: ${designTokens.heightForInput};
+      align-items: center;
+      display: flex;
+      font-size: ${designTokens.fontSizeForInput};
+      font-family: inherit;
+      min-width: ${designTokens.constraint5};
+      transition: border-color ${designTokens.transitionStandard},
+        box-shadow ${designTokens.transitionStandard};
 
-    &:focus-within {
-      border-color: ${designTokens.borderColorForInputWhenFocused};
-      box-shadow: inset 0 0 0 2px ${designTokens.borderColorForInputWhenFocused};
-    }
-    &:focus,
-    &:hover {
-      border-color: ${props.isDisabled ||
-      props.hasError ||
-      props.hasWarning ||
-      props.isReadOnly ||
-      ((props.isOpen || state.isFocused) && !props.isReadOnly)
-        ? ''
-        : designTokens.borderColorForInputWhenFocused};
-    }
-  `;
+      &:hover:not(:focus) {
+        background-color: ${designTokens.backgroundColorForInputWhenHovered};
+      }
```

Comment:
We need this `hover` to be located before the `focus-within & hover` so we can overwrite it.